spec/unit/pops/loaders/loaders_spec.rb

```diff
@@ -234,15 +234,15 @@ def compile_and_get_notifications(code)
     end
 
     it "an autoloaded type can reference an autoloaded type alias from another module when dependency is present in metadata.json" do
-      File.stubs(:read).with(user_metadata_path).returns user_metadata.merge('dependencies' => [ { 'name' => 'test-usee'} ]).to_pson
+      Puppet::FileSystem.stubs(:read).with(user_metadata_path, {:encoding => 'utf-8'}).returns user_metadata.merge('dependencies' => [ { 'name' => 'test-usee'} ]).to_pson
       expect(eval_and_collect_notices(<<-CODE, node)).to eq(['ok'])
         assert_type(User::WithUseeZero, [0])
         notice(ok)
       CODE
     end
 
     it "an autoloaded type can reference an autoloaded type alias from another module when other module has it declared in init.pp" do
```

Comment:
It's likely a religious battle on what constitutes a unit. Let's side step that for a moment. 

> An integration test would require laying out the involved modules, the type aliases in separate files etc. and use no mocking.

Terminology is hard, "integration" is overloaded.  It depends on your definition of "integration testing" - I specifically used the wording "white box integration" (apologies though, it was only in the statement prior to your mention). White box integration means you can still reach into the system and put certain pieces together, use test doubles for other areas of code, and test how interactions work within those integrated parts of the system under test, which is what I'm seeing here in these tests. 

What I'm hearing by your use of "integration" seems more black box integration, acceptance testing and/or end to end testing?

`spec/unit` versus `spec/integration` in this case.

> The same style is used in many places throughout our code base

Agreed, but those other places in the code base may also fall under white box integration and thus are only relevant to the current discussion in that they may also could be subject to a similar discussion eventually.